lib/swiss_schema.ex

```diff
@@ -533,25 +554,33 @@ defmodule SwissSchema do
       end
 
       @impl SwissSchema
-      def insert(%{} = params, opts \\ []) do
+      def insert(source, opts \\ [])
```

Comment:
This isn't ideal. Changing the API might cause bad effects in some code using this library in production.

If we're going to touch the API like that, we need to discuss and plan these changes and their releases accordingly.

For now, let's focus on supporting custom changesets in this PR.